core/src/main/java/com/google/zxing/client/result/ResultParser.java

```diff
@@ -163,26 +165,27 @@ protected static boolean isSubstringOfDigits(CharSequence value, int offset, int
     return value.length() >= max && DIGITS.matcher(value.subSequence(offset, max)).matches();
   }
 
-  static Map<String,String> parseNameValuePairs(String uri) {
+  static String[] parseNameValuePairs(String uri) {
     int paramStart = uri.indexOf('?');
     if (paramStart < 0) {
       return null;
     }
-    Map<String,String> result = new HashMap<>(3);
+    String[] result = new String[3];
```

Comment:
This does not work. 3 is an expected size to the hashmap, but is a fixed size for an array. This will keep overwriting the same value